lua/mapvote/client/vgui/frame.lua

```diff
@@ -1,7 +1,91 @@
+---@class MapVote_Frame : DFrame
 local PANEL = {}
 
+local function disableColor( c )
+    return Color( c.r / 2, c.g / 2, c.b / 2 )
+end
+
+local function drawCircle( x, y, radius, seg )
+    local cir = {}
+
+    table.insert( cir, { x = x, y = y, u = 0.5, v = 0.5 } )
+    for i = 0, seg do
+        local a = math.rad( (i / seg) * -360 )
+        table.insert( cir,
+            {
+                x = x + math.sin( a ) * radius,
+                y = y + math.cos( a ) * radius,
+            } )
+    end
+
+    local a = math.rad( 0 ) -- This is needed for non absolute segment counts
+    table.insert( cir,
+        {
+            x = x + math.sin( a ) * radius,
+            y = y + math.cos( a ) * radius,
+        } )
+
+    surface.DrawPoly( cir )
+end
+function surface.DrawTexturedRectRotatedPoint( x, y, w, h, rot, x0, y0 )
+    local c = math.cos( math.rad( rot ) )
+    local s = math.sin( math.rad( rot ) )
+
+    local newx = y0 * s - x0 * c
+    local newy = y0 * c + x0 * s
+
+    surface.DrawTexturedRectRotated( x + newx, y + newy, w, h, rot )
+end
+
+function PANEL:SetHideOnClose( hide )
+    self.hideOnClose = hide
+end
+
+function PANEL:Init()
```

Comment:
Have you tested other segment amounts for the circles in this function? It can be a major performance sink at larger numbers.